### PR TITLE
Modify determineMouseOver to check SpellFlyout

### PR DESCRIPTION
**Pull request by markoleptic**

### CODE REVIEW

File: BlizzHUDTweaks/MouseoverFrameFading.lua [hunk 1]
```diff
@@ -207,22 +207,49 @@ end
...
-          if addon:GetFrameMapping()[linkedFrameName].mainFrame:IsMouseOver() then
-            linkedFrameMouseover = true
-            return linkedFrameMouseover or frameOptions.mainFrame:IsMouseOver()
+      local frameProfile = profile[linkedFrameName]
+      if frameProfile and frameProfile.Enabled then
+        if linkedFrames[linkedFrameName] then
+          local mappedFrame = addon:GetFrameMapping()[linkedFrameName]
+          if mappedFrame and mappedFrame.mainFrame then
+            if spellFlyoutButtonBarName == mappedFrame.mainFrame:GetName() then
+              return true
```
Comment (PhenomDevel) [inline BlizzHUDTweaks/MouseoverFrameFading.lua:233]: If we determine that a linked frame is being hovered over, we can return it immediately without having to possibly loop through all the other frames as well. Don't we?

File: BlizzHUDTweaks/MouseoverFrameFading.lua [hunk 2]
```diff
@@ -207,22 +207,49 @@ end
...
+            end
+            if mappedFrame.mainFrame:IsMouseOver() then
+              linkedFrameMouseover = true
```
Comment (PhenomDevel) [inline BlizzHUDTweaks/MouseoverFrameFading.lua:236]: Same as above.

File: BlizzHUDTweaks/MouseoverFrameFading.lua [hunk 3]
```diff
@@ -207,22 +207,49 @@ end
...
-  return frameOptions.mainFrame.__BlizzHUDTweaksForceMouseover or frameOptions.mainFrame:IsMouseOver()
+  -- Frame with no links
+  local mappedFrame = addon:GetFrameMapping()[frameName]
+  if mappedFrame and mappedFrame.mainFrame then
+    if spellFlyoutButtonBarName == mappedFrame.mainFrame:GetName() then
+      return true
+    end
+  end
+
+  return linkedFrameMouseover or fallback
```
Comment (PhenomDevel) [inline BlizzHUDTweaks/MouseoverFrameFading.lua:252]: If we immediately  return when a frame/linked frame is determined to be hovered we only need to return the fallback or false

**Review by PhenomDevel — CHANGES_REQUESTED** [review]
Some small questions which when correct should be fixed.

File: BlizzHUDTweaks/MouseoverFrameFading.lua (hunk 1 above)
Comment (PhenomDevel) [reply BlizzHUDTweaks/MouseoverFrameFading.lua:233]: This comment was supposed to be on the line mentioned in the next comment, sorry for the confusion :D

**Comment by markoleptic** [comment]
I tried doing early returns earlier when I was working on the fix, but somehow it caused things to break. However it seems to work fine now. Good suggestion!

**Comment by PhenomDevel** [comment]
I could not find any strange behavior in my tests. I only tested with my mage though. Seems to work fine and is merged. Will be available in 1.49.4 soon.

Thanks for the PR :)

**Pull request merged**